blocks/library/list/index.js

```diff
@@ -15,6 +15,42 @@ import BlockControls from '../../block-controls';
 
 const { children, prop } = hpq;
 
+const fromBrDelimitedContent = function( content ) {
+	const listItems = [];
+	listItems.push( createElement( 'li', [], [] ) );
+	content.forEach( function( element, elementIndex, elements ) {
+		// "split" the incoming content on 'br' elements
+		if ( 'br' === element.type && elementIndex < elements.length - 1 ) {
+			// if is br and there are more elements to come, push a new list item
+			listItems.push( createElement( 'li', [], [] ) );
+		} else {
+			listItems[ listItems.length - 1 ].props.children.push( element );
+		}
+	} );
+	return listItems;
+};
+
+const toBrDelimitedContent = function( values ) {
+	const content = [];
+	values.forEach( function( li, liIndex, listItems ) {
+		if ( Array.isArray( li.props.children ) ) {
```

Comment:
Another minor idea: I think we could potentially condense this code by using the [`Children` API](https://facebook.github.io/react/docs/react-api.html#react.children) in React. Either by making use of `Children.map` or `Children.toArray`, then the conditional checking for the array won't really be necessary.